mne/export/_edf.py

```diff
@@ -198,6 +198,17 @@ def _export_raw(fname, raw, physical_range, add_ch_type):
                     raise RuntimeError(
                         f"Setting patient birth date to {birthday} "
                         f"returned an error")
+            
+            # EDFwriter compares integer encodings of sex and will
+            # raise a TypeError if 'sex' is not in 'subject_info'
+            # as subj_info.get('sex') in this case return None
+            if sex is None:
+                warning_message = "'sex' not specified in " + \
+                    "raw.info.subject_info. Encoding as n/a: 2 " + \
+                    "for EDFwriter."
+                warn(message=warning_message)
+                sex = 0
+
```

Comment:
Rather than this it would probably be better to do below some `if val is not None`. I think this could live in `_try_to_set_value`.

This assumes EDF allows you to write a subset of the patient info...

Reply:
`EDFwriter` defines a standard `self.__plus_gender = 2` at instantiation, so they actually handle subsets just fine. A `if val is not None` guard clause should result in the same as my implementation.

I actually avoided implementing it in `_try_to_set_value` because I thought it was not a general case, but looking through the functions in `EDFwriter` I see several numeric comparisons, so the issue could arise with other parts of the info.

Let me know what you think, and i'll push a change.

Reply:
I can live with either solution. If we go with this one, though, we shouldn't warn -- if the field is missing from the Raw instance then I think it's okay to just have it be `n/a` when written

Reply:
You can have a look at the change. I have generalized it to the cases of name, sex, and handedness for now. If a similar issue arises in the future for other fields, we could generalize it to more/all fields in the EDF header and move to `_try_to_set_value`.